tests/function.rs

```diff
@@ -84,3 +84,24 @@ fn empty_struct_kind_change_is_modification() {
         }
     }
 }
+
+#[test]
+fn is_reported_lexicographically() {
+    let diff: ApiCompatibilityDiagnostics = parse_quote! {
+        {},
+        {
+            pub fn a() {}
+            pub fn z() {}
+        }
+    };
+    assert_eq!(diff.to_string(), "+ a\n+ z\n");
+
+    let diff: ApiCompatibilityDiagnostics = parse_quote! {
+        {},
+        {
+            pub fn z() {}
+            pub fn a() {}
+        }
+    };
+    assert_eq!(diff.to_string(), "+ a\n+ z\n");
+}
```

Comment:
It can be interesting (in a different PR, maybe?) to group these tests in a specific folder that tests the ApiCompatibilityDiagnostics. So we can test the other parts of the cargo-breaking.